certbot/src/certbot/_internal/renewal.py

```diff
@@ -45,7 +45,8 @@
 STR_CONFIG_ITEMS = ["config_dir", "logs_dir", "work_dir", "user_agent",
                     "server", "account", "authenticator", "installer",
                     "renew_hook", "pre_hook", "post_hook", "http01_address",
-                    "preferred_chain", "key_type", "elliptic_curve"]
+                    "preferred_chain", "key_type", "elliptic_curve",
+                    "preferred_profile", "required_profile"]
```

Comment:
Are there any profiles that need to be consistent between renewals? certbot automatically assumes that values set by the user (on the cli, in cli.ini, etc) will override those pulled out of the renewal config. Is there a way that that would be a problem with any profiles, if they don't match?

Reply:
To make sure I understand the question: Say you have previously issued some certificates for `example.com` with `--preferred-profile=tlsserver`, so `"tlsserver"` is stored in their renewal config. Then you run `certbot renew --preferred-profile=default`. Certbot will now attempt to renew the `example.com` certificate using the `default` profile.

From Let's Encrypt's perspective, that's fine; we don't have any rule that, for instance, "Once you've issued for `example.com` using the `tlssserver` profile, all future issuance for `example.com` must use the `tlsserver` profile."

Reply:
Yes, that is the question.

The only thing that comes to mind is maybe something about cached retry-after intervals for a long-lived cert that switches to a short-lived cert. Obviously we don't do that yet at all, but we'll maybe want to be careful about clearing cached data when switching profiles. Or honestly that's probably best to do on renewal anyway. Maybe then it's just making sure that cached fetched data is cleared on renewal, in case of things like profile switch.